lib/langchain/utils/token_length/cohere_validator.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Langchain
+  module Utils
+    module TokenLength
+      #
+      # This class is meant to validate the length of the text passed in to Cohere's API.
+      # It is used to validate the token length before the API call is made
+      #
+
+      class CohereValidator
+        TOKEN_LIMITS = {
+          # Source:
+          # https://docs.cohere.com/docs/models
+          "command-light" => 4096,
+          "command" => 4096,
+          "base-light" => 2048,
+          "base" => 2048,
+          "embed-english-light-v2.0" => 512,
+          "embed-english-v2.0" => 512,
+          "embed-multilingual-v2.0" => 256,
+          "summarize-medium" => 2048,
+          "summarize-xlarge" => 2048
+        }.freeze
+        #
+        # Calculate the `max_tokens:` parameter to be set by calculating the context length of the text minus the prompt length
+        #
+        # @param content [String | Array<String>] The text or array of texts to validate
+        # @param model_name [String] The model name to validate against
+        # @return [Integer] Whether the text is valid or not
+        # @raise [TokenLimitExceeded] If the text is too long
+        #
+        def self.validate_max_tokens!(content, model_name, client)
```

Comment:
This method is already implemented in `BaseClass` so that we don't need to repeat it: https://github.com/andreibondarev/langchainrb/blob/main/lib/langchain/utils/token_length/base_validator.rb#L15

Reply:
looks like I missed the latest code in my local. let me merge the latest code.